.github/main.yml

```diff
@@ -0,0 +1,55 @@
+name: Flutter Build and Release APP
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'  # Trigger on all tags
+
+jobs:
+  build_and_release:
+    name: Build, Upload, and Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Flutter
+        uses: subosito/flutter-action@v2
+        with:
+          flutter-version: 3.19.5
+
+      - name: Get dependencies
+        run: flutter pub get
+
+      - name: Analyze code
+        run: flutter analyze
+
+      - name: Build APK
+        run: flutter build apk --release  # Include --release for release mode
+
+      - name: Upload APK as artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: android-app
+          path: build/app/outputs/apk/release/app-release.apk  # Update path if needed
+
+      - name: Create GitHub Release (if tag)
+        uses: ncipollo/release-action@v1
+        if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/')  # Only on tag push
+        env:
+          GITHUB_TOKEN: ${{ secrets.PAT }}  # Access token with release permissions
```

Comment:
@manuelgeek we can have an Access token with release permissions created in the repository

Reply:
I think default `github.token` should be enough?

Reply:
You're right